### ISSUE TITLE
simulate rendering in RDRAM

### ISSUE THREAD

**Issue by purplemarshmallow** [gonetz/GLideN64#791]
I tried to implement my idea: Use a test color to determine which pixels are modified. Then copy only the modified pixels. Results are a bit disappointing. Does not even good enough for scoreboards in Mario Tennis. Maybe buffer creation with RDRAM content can help. But I see no ideal solution.
![1](https://cloud.githubusercontent.com/assets/7283660/11192647/8b5aae32-8ca2-11e5-992c-e1ba10f552fc.jpg)
![4](https://cloud.githubusercontent.com/assets/7283660/11192651/90f5a220-8ca2-11e5-9f33-70b51b5ee8e8.jpg)
![3](https://cloud.githubusercontent.com/assets/7283660/11192657/9bbf4b34-8ca2-11e5-8783-8a8eefbb4584.jpg)
![6](https://cloud.githubusercontent.com/assets/7283660/11192659/9f1757b8-8ca2-11e5-9589-0e616ac34846.jpg)
![2](https://cloud.githubusercontent.com/assets/7283660/11192664/a6d1e496-8ca2-11e5-95b3-3027c024162b.jpg)
![5](https://cloud.githubusercontent.com/assets/7283660/11192666/a917e55c-8ca2-11e5-95f8-7b115f5015df.jpg)

**Comment by olivieryuyu** [comment]
well the screenshot for Top Gear Hyper Drive is the best I have ever seen for in a hardware plugin. 

Not sure if this idea can be developed or should be abandoned.

@gonetz 

Any thought about this?

**Comment by gonetz** [comment]
Someday I should find some time to investigate this problem. purplemarshmallow's idea is promising. But I have 200+ other bugs.

**Comment by gonetz** [comment]
@purplemarshmallow Do you have any code left for this feature? I tried to get better results with Top Gear Hyper Bike, but with no success.

**Comment by purplemarshmallow** [comment]
I don't know if I saved this code. There were very few lines of code. Mostly an if condition that checks for a certain color before pixels are written to rdram.

I was disappointed because the test color blends with the ground in Top Gear Hyper Bike and I saw no way around it.
Do you need this fearure?

**Comment by gonetz** [comment]
> Do you need this fearure?

Yes, I'd like to look at it. I don't understand what graphics plugin should do to get correct ground textures in Hyper Bike.

**Comment by purplemarshmallow** [comment]
I don't have the old code
I created a branch with this featue
https://github.com/purplemarshmallow/GLideN64/tree/feature_test_color

The code in RDRAMtoColorBuffer.cpp is disabled
Mario Tennis scoreboards are working now without copy from rdram so the feature works

The results in Top Gear Hyperbike are worse now and I don't know why. I failed to compile old versions of the plugin for testing.

**Comment by gonetz** [comment]
I've got the idea, thanks!
